set-dark-theme-variant.py

```diff
@@ -0,0 +1,142 @@
+    win_id = root.get_full_property(NET_ACTIVE_WINDOW,
+                                       Xlib.X.AnyPropertyType).value[0]
+
+    focus_changed = (win_id != last_seen['xid'])
+    if focus_changed:
+        with window_obj(last_seen['xid']) as old_win:
+            if old_win:
+                old_win.change_attributes(event_mask=Xlib.X.NoEventMask)
+
+        last_seen['xid'] = win_id
+        with window_obj(win_id) as new_win:
+            if new_win:
+                new_win.change_attributes(event_mask=Xlib.X.PropertyChangeMask)
+
+    return win_id, focus_changed
+
+def _get_window_name_inner(win_obj):
+    """Simplify dealing with _NET_WM_NAME (UTF-8) vs. WM_NAME (legacy)"""
+    for atom in (NET_WM_NAME, WM_NAME):
+        try:
+            window_name = win_obj.get_full_property(atom, 0)
+        except UnicodeDecodeError:  # Apparently a Debian distro package bug
+            title = "<could not decode characters>"
+        else:
+            if window_name:
+                win_name = window_name.value
+                if isinstance(win_name, bytes):
+                    # Apparently COMPOUND_TEXT is so arcane that this is how
+                    # tools like xprop deal with receiving it these days
+                    win_name = win_name.decode('latin1', 'replace')
+                return win_name
+            else:
+                title = "<unnamed window>"
+
+    return "{} (XID: {})".format(title, win_obj.id)
+
+def get_window_name(win_id):
+    """Look up the window name for a given X11 window ID"""
+    if not win_id:
+        last_seen['title'] = "<no window id>"
+        return last_seen['title']
+
+    title_changed = False
+    with window_obj(win_id) as wobj:
+        if wobj:
+            win_title = _get_window_name_inner(wobj)
+            title_changed = (win_title != last_seen['title'])
+            last_seen['title'] = win_title
+
+    return last_seen['title'], title_changed
+
+def handle_xevent(event):
+    # Loop through, ignoring events until we're notified of focus/title change
+    if event.type != Xlib.X.PropertyNotify:
+        return
+    changed = False
+    try:
+        if event.atom == NET_ACTIVE_WINDOW:
+            if get_active_window()[1]:
+                changed = changed or get_window_name(last_seen['xid'])[1]
+        elif event.atom in (NET_WM_NAME, WM_NAME):
+            changed = changed or get_window_name(last_seen['xid'])[1]
+
+        if changed:
+            return last_seen
+    except:
+        pass
+
+def set_theme_variant(window_info, titles_and_variant):
+    window_titles, variant = titles_and_variant
+    if window_info is not None:
+        title = window_info.get('title')
+        if title and title in window_titles:
+            return set_theme_variant_by_window_title(title, variant)
+    return False
+
+def set_theme_variant_by_window_title(title, variant):
+    try:
+        subprocess.call(['xprop', '-name', title, '-f', '_GTK_THEME_VARIANT', '8u',
```

Comment:
You have the xid so it is odd to call xprop like this.

Reply:
It's done that way so it's the same function to set the window dark if it's detected or in the event that it's not detected in 2 secs and it fallback.